experimental/InjectiveResolutions/src/LocalCohomology.jl

```diff
@@ -0,0 +1,642 @@
+# data structures
+struct SectorLC
+  A::Vector{Vector{Int}}
+  sector::Polyhedron
+  index_vector::Vector{Int}
+  H::Generic.QuotientModule
+end
+mutable struct SectorPartitionLC
+  M::SubquoModule{<:MonoidAlgebraElem}
+  i::Int
+  I::MonoidAlgebraIdeal
+  sectors::Vector{SectorLC}
+  maps::Vector{Tuple{SectorLC,SectorLC,Generic.ModuleHomomorphism}}
+
+  function SectorPartitionLC(M::SubquoModule{<:MonoidAlgebraElem}, i::Int, I::MonoidAlgebraIdeal)
+    return new(
+      M,
+      i,
+      I,
+      Vector{SectorLC}(),
+      Vector{Tuple{SectorLC,SectorLC,Generic.ModuleHomomorphism}}(),
+    )
+  end
+end
+
+@doc raw"""
+    is_zero(s::SectorLC)
+
+Test if local cohomology is zero on a sector.
+"""
+function is_zero(s::SectorLC)
+  return dim(s.H) == 0
+end
+
+@doc raw"""
+    is_zero(S::SectorPartitionLC)
+
+Given a sector partition of a local cohomology module, check if the local cohomology module is zero.
+"""
+function is_zero(S::SectorPartitionLC)
+  return all([dim(s.H) == 0 for s in S.sectors])
+end
+
+function print_A_vector(io::IO, A::Vector{Vector{Int}})
+  print(io, "[")
+  for (i, subvec) in enumerate(A)
+    print(io, isempty(subvec) ? "[]" : string(subvec))
+    if i != length(A)
+      print(io, ", ")
+    end
+  end
+  print(io, "]")
+end
+
+function Base.show(io::IO, ::MIME"text/plain", S::SectorLC)
+  println(
+    io, "sector S_A of ZZ^", ambient_dim(S.sector), " of dimension ", dim(S.sector), " with"
+  )
+  print(io, "   A = ")
+  print_A_vector(io, S.A)
+  print(io, " and local cohomology of dimension ",
+    dim(S.H),
+  )
+end
+
+function Base.show(io::IO, S::SectorLC)
+  print(io, "sector of ZZ^", ambient_dim(S.sector), " with local cohomology of dimension ", dim(S.H))
+end
+
+function Base.show(io::IO, ::MIME"text/plain", SP::SectorPartitionLC)
+  println(
+    io,
+    "sector partition of ",
+    SP.i,
+    "-th local cohomology module supported on ideal (",
+    join(gens(SP.I), ", "),
+    ") of ",
+  )
+  println(io, " ", SP.M)
```

Comment:
Found thanks to your new `jldoctest` which revealed the printing issue ;-)
```suggestion
  print(io, " ", SP.M)
```